provider/resource_keycloak_users_permissions.go

```diff
@@ -0,0 +1,278 @@
+		return err
+	}
+	if permissionView != nil && len(permissionView.Policies) > 0 {
+		data.Set("view_scope_policy_id", permissionView.Policies[0])
+	}
+	permissionManage, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage"].(string))
+	if err != nil {
+		return err
+	}
+	if permissionManage != nil && len(permissionManage.Policies) > 0 {
+		data.Set("manage_scope_policy_id", permissionManage.Policies[0])
+	}
+	permissionMapRoles, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["map-roles"].(string))
+	if err != nil {
+		return err
+	}
+	if permissionMapRoles != nil && len(permissionMapRoles.Policies) > 0 {
+		data.Set("map_roles_scope_policy_id", permissionMapRoles.Policies[0])
+	}
+	permissionManageGroupMembership, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["manage-group-membership"].(string))
+	if err != nil {
+		return err
+	}
+	if permissionManageGroupMembership != nil && len(permissionManageGroupMembership.Policies) > 0 {
+		data.Set("manage_group_membership_scope_policy_id", permissionManageGroupMembership.Policies[0])
+	}
+	permissionImpersonate, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["impersonate"].(string))
+	if err != nil {
+		return err
+	}
+	if permissionImpersonate != nil && len(permissionImpersonate.Policies) > 0 {
+		data.Set("impersonate_scope_policy_id", permissionImpersonate.Policies[0])
+	}
+	permissionUserImpersonated, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions["user-impersonated"].(string))
+	if err != nil {
+		return err
+	}
+	if permissionUserImpersonated != nil && len(permissionUserImpersonated.Policies) > 0 {
+		data.Set("user_impersonated_scope_policy_id", permissionUserImpersonated.Policies[0])
+	}
+	data.Set("authorization_resource_server_id", realmManagementClient.Id)
+
+	return nil
+}
+
+func resourceKeycloakUsersPermissionsDelete(data *schema.ResourceData, meta interface{}) error {
+
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+
+	usersPermissions, err := keycloakClient.GetUsersPermissions(realmId)
+	if err == nil && usersPermissions.Enabled {
+		_ = unsetUsersScopePermissionPolicy(keycloakClient, realmId, "view")
+	}
+	return keycloakClient.DisableUsersPermissions(realmId)
+}
+
+func resourceKeycloakUsersPermissionsImport(d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("Invalid import. Supported import formats: {{realmId}}")
+	}
+	d.Set("realm_id", parts[0])
+
+	d.SetId(parts[0])
+
+	return []*schema.ResourceData{d}, nil
```

Comment:
You indicated that only realm ID is needed for import
so if there is only a realmId? Is it correct that the incoming Id needs to be split in 2 parts, or that there ever be 2 parts?
So I assume this will always return an error?

Reply:
Doyou think I can use "ImportStatePassthrough" ?
https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_importer.go#L47

Reply:
To be honest, I always just copy paste this from other resources. i always forget how this exactly works
lets ask @mrparkers ....

Reply:
honstly, I do the same most of time ;) 

I ask for `ImportStatePassthrough` because it's used for realm import (https://github.com/mrparkers/terraform-provider-keycloak/blob/master/provider/resource_keycloak_realm.go#L91)

Reply:
`ImportStatePassthrough` is used when all you need to do during import is set the ID and nothing more.  The `keycloak_realm` resource uses this because it doesn't need to set any other attributes for the resource besides its ID.

As an example, look at the import function for the `keycloak_openid_client` resource: https://github.com/mrparkers/terraform-provider-keycloak/blob/ba25fb0d2e25f163784610aa28dd646386b8a322/provider/resource_keycloak_openid_client.go#L438-L447

This is not a good candidate for `ImportStatePassthrough` because the `keycloak_openid_client` resource needs both the ID and `realm_id` attributes in order to function correctly.

Does that make sense?

Reply:
I add an import test :) 

With `schema.ImportStatePassthrough` it does'nt work.
With a dedicated function, it works.

Reply:
Right, `ImportStatePassthrough` won't work because you're setting both the resource ID and the `realm_id` attribute.  You can only use `ImportStatePassthrough` to set the ID of the resource.

Reply:
Looking at this again - you actually can use `ImportStatePassthough` here if you want.  Since the only information we need about this resource is the realm ID, which we use as the unique ID for the resource.

This will only work if you handle that edge case within the `resourceKeycloakUsersPermissionsRead` function though.  Importing this resource when using `ImportStatePassthrough` will cause `data.Get("realm_id")` to return an empty string, so you'll have to handle that within the read function to set that attribute using the resources ID.

Or, you can leave this how it is.  Your call!  :smile: